taxi/views.py

```diff
@@ -64,16 +65,26 @@ class CarDetailView(LoginRequiredMixin, generic.DetailView):
 
 class CarCreateView(LoginRequiredMixin, generic.CreateView):
     model = Car
-    fields = "__all__"
+    form_class = CarUpdateForm
     success_url = reverse_lazy("taxi:car-list")
 
 
 class CarUpdateView(LoginRequiredMixin, generic.UpdateView):
     model = Car
-    fields = "__all__"
+    form_class = CarUpdateForm
     success_url = reverse_lazy("taxi:car-list")
 
 
+@login_required
+def car_add_or_remove_driver(request, pk):
+    driver = Driver.objects.get(id=request.user.pk)
+    if driver in Driver.objects.all():
+        driver.cars.remove(pk)
+    else:
+        driver.cars.add(pk)
+    return redirect(reverse_lazy("taxi:car-list"))
```

Comment:
is it works as expected ? U are getting driver by id and then check if it exist in list of all drivers, am i right ? 
fix that, also no need to get driver here, u can use reuqest.user instead